### PR TITLE
Upgrade CNI dependency to v1.1.2 and update all CNI plugins to support spec v1.0.0

### LINKED ISSUE #97: Upgrade CNI dependency to v1.1.2 and update all CNI plugins to support spec v1.0.0
### Summary
Upgrade CNI dependency to v1.1.2 and update all CNI plugins to support spec v1.0.0

### Description
Amazon ECS is upgrading its CNI components across EC2 and Fargate to spec v1.0.0 so that we can build a single networking stack and build process.

### PR DESCRIPTION
**Pull request by ofiliz**
*Issue #, if available:*
#97 

*Description of changes:*
This PR consists of the following set of commits:
* Upgrade CNI dependency to v1.1.2
* Update CNI test netconfig files to spec v1.0.0
* Upgrade CNI plugins to support CNI spec v1.0.0
* Update tests to cover spec v1.0.0 behavior

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

Closes #97

### CODE REVIEW

**Comment by rawahars** [comment]
All the plugins in this repository are only supporting specs `0.3.0` and `0.3.1`. We need to add `0.4.0` and `1.0.0` in there for backward compatibility.

Reference: https://github.com/aws/amazon-vpc-cni-plugins/blob/0b99afc17f52858ed4d7459fdfc01e6ddcb44327/plugins/aws-appmesh/plugin/plugin.go#L32
https://github.com/aws/amazon-vpc-cni-plugins/blob/0b99afc17f52858ed4d7459fdfc01e6ddcb44327/plugins/ecs-serviceconnect/plugin/plugin.go#L31
https://github.com/aws/amazon-vpc-cni-plugins/blob/0b99afc17f52858ed4d7459fdfc01e6ddcb44327/plugins/vpc-branch-eni/plugin/plugin.go#L33
https://github.com/aws/amazon-vpc-cni-plugins/blob/0b99afc17f52858ed4d7459fdfc01e6ddcb44327/plugins/vpc-bridge/plugin/plugin.go#L33
https://github.com/aws/amazon-vpc-cni-plugins/blob/0b99afc17f52858ed4d7459fdfc01e6ddcb44327/plugins/vpc-eni/plugin/plugin.go#L34
https://github.com/aws/amazon-vpc-cni-plugins/blob/0b99afc17f52858ed4d7459fdfc01e6ddcb44327/plugins/vpc-tunnel/plugin/plugin.go#L33

**Comment by ofiliz** [comment]
Yes, those changes are in additional commits that I will submit once these initial two are approved. I won't merge until after all commits are in.

**Review by ohsoo — APPROVED** [review]

**Pull request merged**
